Automattic/sensei
# Improve classic editor support for questions

### Background

The classic editor plugin offers a choice to edit something in the block editor or classic editor. We previously forced falling back to metaboxes for quizzes and questions when the classic editor plugin was active, which breaks the case when the post is edited in the block editor. 

<img width="485" alt="image" src="https://user-images.githubusercontent.com/176949/183861045-d7fa06dc-4c16-4da7-8a51-665b5310cdac.png">

Some new question types only work with the block editor, so this enables the option to create/edit those with the block editor and use the classic editor for other cases. 

### Changes proposed in this Pull Request

* Change check for block/metabox-based question/quiz editor to be based on the current screen being the block editor
* Add support for Question post types to be opened in the classic or the block editor
* Remove `is_legacy_quiz_editor` tracking flag as it's more nuanced now — having the classic editor plugin on doesn't mean it's used for editing everything

### Testing instructions

* Add the classic editor plugin
* Try to edit questions and lessons with quizzes both in classic and block editor. Make sure both work.
